base/workerpool.jl

```diff
@@ -114,3 +117,57 @@ using `remotecall_fetch`.
 """
 remote(f) = (args...; kwargs...)->remotecall_fetch(f, default_worker_pool(), args...; kwargs...)
 remote(p::WorkerPool, f) = (args...; kwargs...)->remotecall_fetch(f, p, args...; kwargs...)
+
+type CachedWorkerPool # Manages a cache for code/data on workers.
+    pool::WorkerPool
+    directory::Dict{Tuple, RemoteChannel}  # Mapping between a tuple (worker_id, object_id) and a remote_ref
+    CachedWorkerPool(pool) = new(pool, Dict{Int, Any}())
+end
+
+function release(p::CachedWorkerPool)  # For a quick release of remote function cache
+    for (_,rr) in p.directory
+        finalize(rr)
+    end
+    empty!(p.directory)
+end
+
+function exec_from_cache(f, rr::RemoteChannel, args...)
+    if (f==nothing)
```

Comment:
no parens around `if` condition, usually a bit better to compare to `nothing` using `===`